### PR TITLE
I restructured Readme.md, simplified samples

### PR DESCRIPTION
**Pull request by vit100**
Idea of this was to make clear sections
1. Config
1.1 with config.json file.
1.2 without (explicit config)
2. Usage: 
2.1PKCE way
2.2 Implicit flow 

Plus rest of docs.

Please look at final file here: [https://github.com/vit100/angular-auth-oidc-client/blob/master/README.md](https://github.com/vit100/angular-auth-oidc-client/blob/master/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,26 +107,111 @@ export function loadConfig(oidcConfigService: OidcConfigService) {
...
+	"forbidden_route": "/forbidden",
+	"unauthorized_route": "/unauthorized",
+	"log_console_warning_active": true,
+	"log_console_debug_active": true,
+  "max_id_token_iat_offset_allowed_in_seconds": 10,
+}
+```
+At present only the 'code' with PKCE, 'id_token token' or the 'id_token' flows are supported:
+
+`"response_type": ["code" | "id_token token" | "id_token" | "token" ]`
```
Comment (damienbod) [inline README.md]: can you remove the "token", this is not supported

Comment (damienbod) [reply README.md]: `"response_type": ["code" | "id_token token" | "id_token"  ]`

File: README.md [hunk 2]
```diff
@@ -101,26 +107,111 @@ export function loadConfig(oidcConfigService: OidcConfigService) {
...
         this.oidcConfigService.onConfigurationLoaded.subscribe(() => {
-            const config = new OpenIDImplicitFlowConfiguration();
-            config.stsServer = 'https://your_secure_token_service_url';
-            config.redirect_url = 'https://localhost:4200';
-            config.client_id = 'angular_code_client';
-            config.scope = 'openid profile email';
-            config.response_type = 'code';
+          const config = new OpenIDImplicitFlowConfiguration();
+          //merge configuration loaded from assets/auth.clientConfiguration.json
+          Object.assign(implicitFlowConfig, this.oidcConfigService.clientConfiguration);
```
Comment (damienbod) [inline README.md]: do you need to set the config object here?

Comment (damienbod) [reply README.md]: config  instead of implicitFlowConfig?

Comment (FabianGosebrink) [inline README.md]: Shouldn't this be `Object.assign(config, this.oidcConfigService.clientConfiguration);`?

File: README.md (hunk 1 above)
Comment (vit100) [reply README.md]: done.

File: README.md (hunk 2 above)
Comment (vit100) [reply README.md]: done

**Comment by vit100** [comment]
Cool.


I'm planning to complete online sample with implicit flow accompanying this document. I will need stsServer with demo credentials. 


Can I use yours (somewhere on AAD)? If yes, u will need to guide me demo username, pass, client ID, and I will confirm callback later..


Any thoughts?




On Sun, Apr 14, 2019 at 9:21 AM -0400, "damienbod" <notifications@github.com> wrote:












Merged #429 into master.

**Pull request merged**